engine-kommand/src/commonMain/kotlin/com/saveourtool/processbuilder/KommandEngine.kt

```diff
@@ -0,0 +1,66 @@
+package com.saveourtool.processbuilder
+
+import com.saveourtool.processbuilder.utils.CurrentOs
+import com.saveourtool.processbuilder.utils.cwd
+import com.saveourtool.processbuilder.utils.getCurrentOs
+import com.saveourtool.processbuilder.utils.logger
+import com.saveourtool.processbuilder.utils.stderr
+import com.saveourtool.processbuilder.utils.stdout
+
+import com.kgit2.process.Command
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+
+private typealias CommandWithArgs = Pair<String, Array<String>>
+
+object KommandEngine : Engine {
+    private val scope = CoroutineScope(Dispatchers.Default)
+
+    override suspend fun execute(
+        command: String,
+        config: ProcessBuilderConfig,
+    ): ExecutionResult {
+        val (cmd, args) = getCommandWithArgs(escapeDoubleTicks(command)).also {
+            logger.info { "${it.first} ${it.second.joinToString(" ")}" }
+        }
+        val stdout = config.stdout
+        val stderr = config.stderr
+        val child = Command(cmd)
```

Comment:
should this command be created inside `async` block? because now we create everythin and only then run a `scope.asycn` to get a result

Reply:
Now `execute` is `suspend`, so I use the scope of a caller. `stdout` and `stderr` are read with `produce` builder over `Dispatchers.IO` (it is now available on Kotlin Native). So now there are no `async` invocations at all